### ISSUE TITLE
[bug] cohort maxRelativeDelta tolerance is applied only to errorRate, ignored for all other metrics

### ISSUE THREAD

**Issue by abhay-codes07** [rajudandigam/agent-inspect#120]
## Summary

`tolerance.maxRelativeDelta` in cohort analysis is silently ignored for every numeric metric except `errorRate`. The option is documented generically in `packages/core/src/cohort/types.ts` ("Allowed relative delta (0-1) before a metric comparison is a regression") and `analyzeCohort` threads it through for all metrics, but in `packages/core/src/cohort/compare.ts` only the `errorRate` case passes it to `compareNumber`. The `duration`, `llmCallCount`, `tokenUsage`, `retryCount`, `observationFailure`, `guardrailFailure`, `circuitViolation`, and `redactionWarning` cases call `compareNumber` without the tolerance argument, so the tolerance check never runs for them.

Effect: a within-tolerance change on any of those metrics is still flagged as a regression, flipping `result.ok` to false. A 1% duration increase with `maxRelativeDelta: 0.1` (10% allowed) fails the cohort comparison.

## AgentInspect version

main at 1c63c55 (6.7.2)

## Environment

- Node.js version: 24.11.0
- OS: Windows 11 (not OS specific)
- Package manager (npm/pnpm/yarn): pnpm 9.15.0 (repo pin)

## Steps to reproduce

```ts
import { compareCohortAggregates } from "agent-inspect/advanced";

const comparisons = compareCohortAggregates(
  [
    { groupKey: "all", cohortLabel: "before", runCount: 10, errorRate: 0, avgDurationMs: 100, avgLlmCallCount: 0, avgRetryCount: 0, observationFailureRate: 0, avgGuardrailFailures: 0, avgCircuitViolations: 0, avgRedactionWarnings: 0 },
    { groupKey: "all", cohortLabel: "after",  runCount: 10, errorRate: 0, avgDurationMs: 101, avgLlmCallCount: 0, avgRetryCount: 0, observationFailureRate: 0, avgGuardrailFailures: 0, avgCircuitViolations: 0, avgRedactionWarnings: 0 },
  ],
  { baseline: "before", candidate: "after", metrics: ["duration"], maxRelativeDelta: 0.1 },
);
console.log(comparisons[0].regression);
```

## Expected behavior

`false`: the 1% delta is inside the 10% allowed relative delta, same as the equivalent `errorRate` comparison.

## Actual behavior

`true`: the duration case never receives the tolerance, so any positive delta is a regression regardless of `maxRelativeDelta`.

## Trace / log sample

Synthetic aggregate objects only, shown inline above.

## Product boundary check

- [x] This is not a request for hosted observability, vendor upload, replay, or cost analytics (see [docs/LIMITATIONS.md](../../docs/LIMITATIONS.md))

## Additional context

`packages/core/test/cohort/cohort.test.ts` never passes `tolerance`, so the asymmetry is untested. The fix is mechanical: pass the tolerance to every `compareNumber` call (the categorical `toolChoice`/`toolOrdering` comparisons are unaffected since a relative delta has no meaning there). I have the fix plus a per-metric regression matrix ready and will open a PR referencing this issue.

**Comment by abhay-codes07** [comment]
Fix PR is up with a per-metric tolerance matrix. No-tolerance behavior is byte-identical, so existing callers are unaffected.
